### PR TITLE
Fix Index mul/div ops with Series, closes #19080, #19042

### LINKED ISSUE #19042: Index * Series returns Index
Opening mainly so I have something to reference in the upcoming PR.

```
tdi = pd.TimedeltaIndex(['0days', '1day', '2days', '3days', '4days'])
ser = Series([0, 1, 2, 3, 4], dtype=np.int64)
expected = Series(['0days', '1day', '4days', '9days', '16days'],
                          dtype='timedelta64[ns]')

result = tdi * ser
tm.assert_series_equal(result, expected)   # --> fails

>>> result
TimedeltaIndex(['0 days', '1 days', '4 days', '9 days', '16 days'], dtype='timedelta64[ns]', freq=None)
```

Same for division:
```
tdi = pd.TimedeltaIndex(['0days', '1day', '2days', '3days', '4days'])
ser = Series([1.5, 3, 4.5, 6, 7.5], dtype=np.float64)
expected = Series([tdi[n] / ser[n] for n in range(len(ser))], dtype='timedelta64[ns]')

result = tdi / ser
>>> result
TimedeltaIndex(['00:00:00', '08:00:00', '10:40:00', '12:00:00', '12:48:00'], dtype='timedelta64[ns]', freq=None)
```

### LINKED ISSUE #19080: Index * Series returns Index
Holdover from #19042 (see discussion in #19044), the example below applies to regular Index objects too.  Easy fix (tag as good-first-PR?), just need to add a check [here](https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexes/base.py#L4022) that returns `NotImplemented` if `isinstance(other, ABCSeries)`.  Updating the appropriate tests is the labor-intensive part.

```
tdi = pd.TimedeltaIndex(['0days', '1day', '2days', '3days', '4days'])
ser = Series([0, 1, 2, 3, 4], dtype=np.int64)
expected = Series(['0days', '1day', '4days', '9days', '16days'],
                          dtype='timedelta64[ns]')

result = tdi * ser
tm.assert_series_equal(result, expected)   # --> fails

>>> result
TimedeltaIndex(['0 days', '1 days', '4 days', '9 days', '16 days'], dtype='timedelta64[ns]', freq=None)
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #19080, closes #19042
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
lgtm. pls add a whatsnew note, bug fix / conversion. in generall *anything* that is user facing needs a whatsnew note. ping on green.

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -423,6 +423,9 @@ Conversion
 - Bug in :class:`WeekOfMonth` and :class:`LastWeekOfMonth` where default keyword arguments for constructor raised ``ValueError`` (:issue:`19142`)
 - Bug in localization of a naive, datetime string in a ``Series`` constructor with a ``datetime64[ns, tz]`` dtype (:issue:`174151`)
 - :func:`Timestamp.replace` will now handle Daylight Savings transitions gracefully (:issue:`18319`)
+- Bug in :class:`Index` multiplication and division methods where operating with a ``Series`` would return an ``Index`` object instead of a ``Series`` object (:issue:`19042`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: in another PR, pls separate out all of the ops changes into a separate sub-section of Bug Fixes (as I think we have a lot), call it Operations

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

File: doc/source/whatsnew/v0.23.0.txt (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v0.23.0.txt]: Sure.  All 4 current entries in the Numeric subsection could plausibly belong in Operations.  Move them or leave them be?

Comment (jbrockmendel) [reply doc/source/whatsnew/v0.23.0.txt]: Also a) also move comparison method changes or just arithmetic methods and b) also move offset arithmetic, or just Series/Index?

**Pull request merged**